Fix: don't annoy when java tmpdir jvmarg is set & /tmp is noexec

turret-dev opened the pull request:
Got annoyed, didn't want to see it anymore.

Sorry for using goto but I didn't see a way to skip the syscalls without adding for(;0;), do {}, or any other weird symantics when goto seemed much simpler.

Also: changed the java tmpdir noexec warning to include information about remedying the problem, including remounting /tmp as 'exec' and setting the java.io.tmpdir jvm arg which solves the issue.

First contribution, please let me know if I didn't follow some code style or weird Qt/CPP coding recommendations (I'm a C programmer by trade)

Trial97 commented:
Suggestion to avoid go to just wrap the block that starts at 1028 inside the negation of the JVM arguments that you have(aka move the arguments before 1028 ans at 1028 put if condition to check tmp did do check

turret-dev commented:
> Suggestion to avoid go to just wrap the block that starts at 1028 inside the negation of the JVM arguments that you have(aka move the arguments before 1028 ans at 1028 put if condition to check tmp did do check

imo that just adds additional nesting notwithstanding the already nested code. if you want i can but im just against it, not like goto is super evil and a criminal

Trial97 commented:
Ok evil no but I do believe that is not needed.
Yes I'm aware of nesting hell but is better than goto. If you want to not be a nesting hell just move that block into a function and do early returns.

turret-dev commented:
> Ok evil no but I do believe that is not needed. Yes I'm aware of nesting hell but is better than goto. If you want to not be a nesting hell just move that block into a function and do early returns.

ok i fix

Kationor commented on launcher/Application.cpp:
```diff
@@ -1045,7 +1046,11 @@ Application::Application(int& argc, char** argv) : QApplication(argc, argv)
-                   "Some versions of Minecraft may not launch.\n");
+                   "Some versions of Minecraft may not launch.\n"
+                   "\n"
+                   "You may solve this issue by remounting /tmp as 'exec', or alternatively "
+                   "you may set the java.io.tmpdir JVM arg to a writeable directory in a "
+                   "filesystem where the 'exec' flag is set (i.e., /home/user/.local/tmp)\n");
```
```suggestion
                   "You may solve this issue by remounting /tmp as 'exec' or setting"
                   "the java.io.tmpdir JVM argument to a writeable directory in a "
                   "filesystem where the 'exec' flag is set (e.g., /home/user/.local/tmp)\n");
```

turret-dev replied:
thanks for the better wording!

Kationor replied:
Looks like you didn't apply the change properly. I also just noticed that I forgot a trailing space on line 1051, so might wanna fix that while you're at it.

turret-dev replied:
yup, fixed that, forgot to run `git add` on the changes. good reminder that you should check stuff before you commit it.. :)

turret-dev committed:
fix: don't annoy when java tmpdir jvmarg is set & /tmp is noexec

Signed-off-by: Mason Rocha <mrocha@turretllc.us>

Got annoyed, didn't want to see it anymore.

Java tmpdir noexec warning modified to include information about
possible remedies available
